coffee-chats/src/main/webapp/src/util/fetch.ts

```diff
@@ -4,9 +4,12 @@ import React from "react";
  * A hook that fetches JSON data from a URL using a GET request
  *
  * @param url: URL to fetch data from
+ * @param initial: Initial value, returned when no data was fetched yet
+ * @returns a pair: data and function that causes the data to be fetched again when called
  */
-export function useFetch(url: string): any {
-  const [data, setData] = React.useState(null);
+export function useFetch(url: string, initial: any = null): [any, () => void] {
+  const [data, setData] = React.useState(initial);
+  const [forceUpdate, setForceUpdate] = React.useState(0);
```

Comment:
This setForceUpdate is adding a lot of complexity to this function. If I understand correctly, the idea is that performing the fetch would ideally automatically re-render with the new value that is set? And that the current data is being held in a persistent variable using data which is being maintained for every single call to fetch?

In fact it seems like you're trying to do a way of doing some sort of central state management here that updates when called, which I do think is the correct idea! But I think perhaps we should look into using something like Redux to handle this behaviour.

Reply:
`useFetch` is a hook, it doesn't manage any central state, whatever state it has is local to the call to it within a component. But I guess we could use something like https://use-http.com instead (which is the same thing really, just a bit more functional).

The idea is for `useFetch` to return a pair (like `useState` does): current value fetched from remote and a function, that when called will fetch that value again and force the component to rerender.

Reply:
I think I understand it correctly now in terms of the comparison to how useState functions. My apologies I misunderstood how the useState calls would work from different components before, but yes they should be fine and individual to each call.